src/main/java/org/apache/hadoop/mapred/MesosExecutor.java

```diff
@@ -51,7 +51,7 @@ private JobConf configure(final TaskInfo task) {
 
     // Set the mapred.local directory inside the executor sandbox, so that
     // different TaskTrackers on the same host do not step on each other.
-    conf.set("mapred.local.dir", System.getProperty("user.dir") + "/mapred");
+    conf.set("mapred.local.dir", System.getenv("MESOS_DIRECTORY") + "/mapred");
```

Comment:
This is a really great change, never noticed it was behaving like this before :smile: